src/facade/dragonfly_connection.cc

```diff
@@ -469,10 +469,17 @@ bool Connection::IsAdmin() const {
   return static_cast<Listener*>(owner())->IsAdminInterface();
 }
 
+bool Connection::IsMain() const {
+  return static_cast<Listener*>(owner())->IsMainInterface();
+}
+
 io::Result<bool> Connection::CheckForHttpProto(FiberSocketBase* peer) {
-  bool primary_port_enabled = absl::GetFlag(FLAGS_primary_port_http_enabled);
-  bool admin = IsAdmin();
-  if (!primary_port_enabled && !admin) {
+  if (!IsAdmin() && !IsMain()) {
```

Comment:
IsPrivileged  if are adding it

Reply:
I haven't forgotten you, I just didn't get a chance to polish yesterday!